homeassistant/components/upcloud/config_flow.py

```diff
@@ -0,0 +1,111 @@
+
+
+class UpCloudConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """UpCloud config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    username: str
+    password: str
+
+    async def async_step_user(self, user_input=None):
+        """Handle user initiated flow."""
+        if user_input is None:
+            return await self._async_show_form(step_id="user")
+
+        await self.async_set_unique_id(user_input[CONF_USERNAME])
+
+        manager = upcloud_api.CloudManager(
+            user_input[CONF_USERNAME], user_input[CONF_PASSWORD]
+        )
+
+        errors = {}
+        try:
+            await self.hass.async_add_executor_job(manager.authenticate)
+        except upcloud_api.UpCloudAPIError:
+            errors["base"] = "invalid_auth"
+            _LOGGER.debug("invalid_auth", exc_info=True)
+        except requests.exceptions.RequestException:
+            errors["base"] = "cannot_connect"
+            _LOGGER.debug("cannot_connect", exc_info=True)
+
+        if errors:
+            return await self._async_show_form(
+                step_id="user", user_input=user_input, errors=errors
+            )
+
+        return self.async_create_entry(title=user_input[CONF_USERNAME], data=user_input)
+
+    async def async_step_import(self, user_input=None):
+        """Handle import initiated flow."""
+        await self.async_set_unique_id(user_input[CONF_USERNAME])
+        self._abort_if_unique_id_configured()
+
+        return await self.async_step_user(user_input=user_input)
+
+    async def _async_show_form(self, step_id, user_input=None, errors=None):
+        """Show our form."""
+        if user_input is None:
+            user_input = {}
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_USERNAME, default=user_input.get(CONF_USERNAME, "")
+                    ): str,
+                    vol.Required(
+                        CONF_PASSWORD, default=user_input.get(CONF_PASSWORD, "")
+                    ): str,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get options flow."""
+        return UpCloudOptionsFlow(config_entry)
+
+
+class UpCloudOptionsFlow(config_entries.OptionsFlow):
+    """UpCloud options flow."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Handle options flow."""
+
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        data_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_SCAN_INTERVAL,
```

Comment:
Why would we want the scan interval to be configurable ?

Reply:
As far as this PR goes, in order to preserve what we already supported before this change. TBH, I think a configurable scan interval would be good to have for just about everything that needs to be polled, and would be cool if we supported it without having to code anything.

That said, I don't insist keeping the option around here.

Reply:
Yeah a generic scan interval could be nice, but generally we want developers to pick the best scan interval that works.

Reply:
Scan interval requirements may depend on use cases, and developer chosen values cannot work optimally everywhere. Sometimes more frequent updates are needed, sometimes less frequent ones would be fine (or even desirable).

Because we have the option already implemented and working here (and have had it from the beginning of this integration IIRC), I think it would be a shame to drop it for no particularly strong reason.